packages/payment-processor/src/payment/erc20.ts

```diff
@@ -86,10 +140,67 @@ export async function approveErc20(
 }
 
 /**
- * Encodes the approval call, can be used with a Multisig contract.
+ * Processes the approval transaction of a given payment ERC20 to be spent by the swap router,
+ * if the current approval is missing or not sufficient.
+ * @param request request to pay, used to know the network
+ * @param ownerAddress address of the payer
+ * @param paymentCurrency ERC20 currency used for the swap
+ * @param signerOrProvider the web3 provider. Defaults to Etherscan.
+ * @param minAmount ensures the approved amount is sufficient to pay this amount
+ * @param overrides optionally, override default transaction values, like gas.
+ */
+export async function approveErc20ForSwapToPayIfNeeded(
```

Comment:
As I mentioned before, I think this and the next function (among other swap specific code) deserve their own module.